text/0083-multi-level-meta-data-space.md

```diff
@@ -0,0 +1,52 @@
+# Multi-level Meta Data Space
+
+## Summary
+
+PD can handle multiple users' meta data, each user only see its own meta data.
+
+## Motivation
+
+PD is the mata data center of TiKV Cluster, the meta data contains TiKV instances' addr, user data range(region)
+information, the number of replicas of data, etc.
+
+But PD currently only can handle the key space [min-key, max-key], it means that PD only can handle one user's 
+meta data, it is not possible when multiple users want share the same PD cluster as their meta system. There are
+two different scenarios that require PD can handle multiple user' meta data:
+
+1. Multiple TiKV Cluster share the same PD cluster. Because the minimal demplyment of a TiKV Cluster is 3 TiKV 3 PD,
+but it is not cost-effect if every small cluster has 3 dedicated meta data node.
+2. There are Multiple tenant in the same TiKV Cluster, each tenant has it own meta data, each tenant's key range can
+contains any key in the range of [min-key, max-key].
```

Comment:
To make it practical, every APIs need to be accept a `user` prefix. And each users' data can't be stored in the same rocksdb obviously. This also requires PD have knowledge about the underlying storage engine and avoid scheduling replicas from different users to the same storage engine. And TiKV needs to split all memory meta to different users. For example, the index of range becomes `HashMap<UserKey, BTreeMap<Vec<u8>, u64>>`.

In my opinion, using prefix is more straightforward and simpler.

Reply:
Using prefix can also achieve the same improvement. The difference of using prefix and using a separate explicit meta is that PD/TiKV/TiDB needs to take good care about meta in the later case.